tests/kafkatest/tests/produce_consume_validate.py

```diff
@@ -28,16 +30,32 @@ class ProduceConsumeValidateTest(Test):
 
     def __init__(self, test_context):
         super(ProduceConsumeValidateTest, self).__init__(test_context=test_context)
+        # How long to wait for the producer to declare itself healthy? This can
+        # be overidden by inheriting classes.
+        self.producer_start_timeout_sec = 60
+
+        # How long to delay between the start of the producer and consumer? This
+        # is important in the case when the consumer is starting from the end,
+        # and we don't want it to miss any messages. The race condition this
+        # timeout avoids is that the consumer is still starting after the
+        # producer begins producing messages, in which case we will miss the
+        # initial set of messages and get spurious test failures.
+        self.delay_between_consumer_and_producer_start_sec = 0
```

Comment:
You're effectively adding a parametrizable `time.sleep` here, and we want to avoid calls to `sleep`.

It's generally better practice to do a `wait_until` and specify the desired state you want to reach before proceeding.

My understanding is that purpose of this is to guarantee that consumer process is "fully alive and ready to consume" (and this probably doesn't require 10 seconds) before continuing on to start the producer. Can you guarantee this with a `wait_until` with a heuristic check on the state of the consumer process rather than a `time.sleep` call?